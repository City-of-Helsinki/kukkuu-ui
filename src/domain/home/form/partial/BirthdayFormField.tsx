import React from 'react';
import { Field } from 'formik';

import InputField from '../../../../common/components/form/fields/input/InputField';
import { formatMessage } from '../../../../common/translation/utils';
import styles from './birthdayFormField.module.scss';

export default function BirthdayFormField() {
  return (
    <div className={styles.birthdayField}>
      <label>
        {formatMessage('homePage.preliminaryForm.childBirthDay.input.label')}
      </label>
      <div className={styles.inputWrapper}>
        <Field
          type="number"
          name="childBirthdayDay"
          component={InputField}
          placeholder={formatMessage(
            'homePage.preliminaryForm.childBirthDay.input.day.placeholder'
          )}
          min={1}
          max={31}
        />
        <div className={styles.dot}>.</div>
        <Field
          type="number"
          name="childBirthdayMonth"
          component={InputField}
          placeholder={formatMessage(
            'homePage.preliminaryForm.childBirthDay.input.month.placeholder'
          )}
          min={1}
          max={12}
        />
        <div className={styles.dot}>.</div>
        <Field
          type="number"
          name="childBirthdayYear"
          component={InputField}
          placeholder={formatMessage(
            'homePage.preliminaryForm.childBirthDay.input.year.placeholder'
          )}
          min={2019}
        />
      </div>
      <small></small>
    </div>
  );
}
