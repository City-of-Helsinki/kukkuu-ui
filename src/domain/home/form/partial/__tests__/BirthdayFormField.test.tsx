import React from 'react';
import { shallow } from 'enzyme';
import { FieldArray } from 'formik';

import BirthdayFormField from '../BirthdayFormField';
import TestForm from '../../../../../common/test/TestForm';

it('renders snapshot correctly', () => {
  const element = shallow(
    <TestForm>
      {() => (
        <FieldArray
          name="foo"
          render={props => <BirthdayFormField {...props} />}
        />
      )}
    </TestForm>
  );
  expect(element.html()).toMatchSnapshot();
});
