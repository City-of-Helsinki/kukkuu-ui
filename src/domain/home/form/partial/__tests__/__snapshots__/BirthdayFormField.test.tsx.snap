// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `"<div class=\\"birthdayField\\"><label>Lapsen syntym√§aika</label><div class=\\"inputWrapper\\"><div class=\\"inputField\\"><div class=\\"inputWrapper numberInput\\"><div class=\\"inputTypeWrapper\\"><input type=\\"number\\" name=\\"child.birthday.day\\" placeholder=\\"pv\\" min=\\"1\\" max=\\"31\\"/></div></div><div class=\\"inputErrorMessage\\"></div></div><div class=\\"dot\\">.</div><div class=\\"inputField\\"><div class=\\"inputWrapper numberInput\\"><div class=\\"inputTypeWrapper\\"><input type=\\"number\\" name=\\"child.birthday.month\\" placeholder=\\"kk\\" min=\\"1\\" max=\\"12\\"/></div></div><div class=\\"inputErrorMessage\\"></div></div><div class=\\"dot\\">.</div><div class=\\"inputField\\"><div class=\\"inputWrapper numberInput\\"><div class=\\"inputTypeWrapper\\"><input type=\\"number\\" name=\\"child.birthday.year\\" placeholder=\\"vuosi\\" min=\\"2019\\"/></div></div><div class=\\"inputErrorMessage\\"></div></div></div><div class=\\"error\\"></div></div>"`;
